src/cargo_ops/elaborate_workspace.rs

```diff
@@ -301,4 +319,102 @@ impl<'ela> ElaborateWorkspace<'ela> {
+                    let dependency = &self.pkg_deps[parent][pkg];
+                    let label = if self.workspace_mode
+                        || parent == &self.workspace.current()?.package_id()
+                    {
+                        pkg.name().to_string()
+                    } else {
+                        format!("{}->{}", self.pkgs[parent].name(), pkg.name())
+                    };
+
+                    let line = Metadata {
+                        name: label,
+                        project: pkg.version().to_string(),
+                        compat: status.compat.to_string(),
+                        latest: status.latest.to_string(),
+                        kind: format!("{:?}", dependency.kind()).to_string(),
+                        platform: dependency
+                            .platform()
+                            .map(|p| p.to_string())
+                            .unwrap_or_else(|| "---".to_owned()),
+                    };
+                    if !crate_graph.dependencies.contains(&line) {
+                        crate_graph.dependencies.insert(line);
+                    }
+                } else {
+                    let line = Metadata {
+                        name: pkg.name().to_string(),
+                        project: pkg.version().to_string(),
+                        compat: status.compat.to_string(),
+                        latest: status.latest.to_string(),
+                        kind: String::new(),
+                        platform: String::new(),
+                    };
+                    if !crate_graph.dependencies.contains(&line) {
+                        crate_graph.dependencies.insert(line);
+                    }
+                }
+            }
+            // next layer
+            if options.flag_depth.is_none() || depth < options.flag_depth.unwrap() {
+                self.pkg_deps[pkg]
+                    .keys()
+                    .filter(|dep| !path.contains(dep))
+                    .filter(|dep| {
+                        !self.workspace_mode
+                            || !self.workspace.members().any(|mem| &mem.package_id() == dep)
+                    })
+                    .for_each(|dep| {
+                        let mut path = path.clone();
+                        path.push(dep);
+                        queue.push_back(path);
+                    });
+            }
+        }
+
+        if crate_graph.dependencies.is_empty() {
+            if !self.workspace_mode {
+                println!("All dependencies are up to date, yay!");
+            }
+        } else {
+            if preceding_line {
+                println!();
+            }
+            
+            let dependencies =  serde_json::to_string(&crate_graph)?;
+            //getting fancy here so we can add in the crate name at the top level for scripting
+            let json = format!("{{\"crate\": \"{}\",{}}}", root.name(), &dependencies[1..dependencies.len()-1]);
```

Comment:
You could wrap the crate graph in a `Serialize`able struct to avoid fudging this JSON bit manually.